ViSearchSDK/ViSearchSDK/Classes/Response/ProductSearch/ViProductSearchResponse.swift

```diff
@@ -0,0 +1,206 @@
+//
+//  ViProductSearchResponse.swift
+//  ViSearchSDK
+//
+//  Created by visenze on 8/3/21.
+//
+
+import Foundation
+
+/// Response parameters from the search requests made
+open class ViProductSearchResponse : NSObject {
+    
+    public var requestId : String? = nil
+    
+    public var status : String? = nil
+    
+    public var imageId : String? = nil
+    
+    public var page : Int? = nil
+    
+    public var limit : Int? = nil
+    
+    public var total : Int? = nil
+    
+    public var error : ViErrorMsg? = nil
+    
+    public var productTypes : [ViProductType] = []
+    
+    public var result : [ViProduct] = []
+    
+    public var catalogFieldsMapping : [String:String] = [:]
+    
+    public var facets : [ViFacet] = []
+    
+    public var productInfo : [String:Any] = [:]
+    
+    public var objects : [ViObjectResult] = []
+    
+    public var groupResults : [ViGroupResult] = []
+    
+    public var groupByKey : String? = nil
+    
+    public var querySysMeta : [String:String] = [:]
+    
+    /// Constructor, uses the raw response from the URL query and parses it into the relevant data fields
+    ///
+    /// - parameter response: Response gotten from the URL request
+    /// - parameter data: Data attached to the response from the server
+    public init(response: URLResponse, data: Data) {
+        super.init()
+        
+        do{
+            let json = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as! Dictionary<String, Any>
+            
+            requestId = json["reqid"]  as? String
+            status    = json["status"] as? String
+            imageId   = json["im_id"]  as? String
+            page      = json["page"]   as? Int
+            limit     = json["limit"]  as? Int
+            total     = json["total"]  as? Int
+            
+            if let err = json["error"] as? [String:Any] {
+                error = ViProductSearchResponse.parseErrorMsg(err: err)
+            }
+            
+            if let pTypesJson = json["product_types"] as? [Any] {
+                productTypes = ViResponseData.parseProductTypes(pTypesJson)
+            }
+            
+            if let results = json["result"] as? [Any] {
+                result = ViProductSearchResponse.parseProductResults(results)
+            }
+            
+            if let mappings = json["catalog_fields_mapping"] as? [String:String] {
+                catalogFieldsMapping = mappings
+            }
+            
+            if let facetListJson = json["facets"] as? [Any] {
+                facets = ViResponseData.parseFacets(facetListJson)
+            }
+            
+            if let prodInfo = json["product_info"] as? [String:Any] {
+                productInfo = prodInfo
+            }
+            
+            if let objs = json["objects"] as? [Any] {
+                objects = ViResponseData.parseObjectResults(objs)
```

Comment:
cannot reuse this method, the result format between visearch and product search are different

Reply:
this object results need to call parseProductResults internally